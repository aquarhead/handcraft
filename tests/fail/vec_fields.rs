use handcraft::handcraft;

#[derive(Debug, Default, PartialEq)]
struct Foo {
  a: u64,
  b: u64,
}

#[derive(Debug, Default, PartialEq)]
struct Baz {
  v: Vec<Foo>,
}

fn main() {
  let crafted = handcraft!(Baz { v: vec![Foo { a: 0 }] });
}
