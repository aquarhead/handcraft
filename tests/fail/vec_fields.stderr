error[E0063]: missing field `b` in initializer of `Foo`
  --> $DIR/vec_fields.rs:15:42
   |
15 |   let crafted = handcraft!(Baz { v: vec![Foo { a: 0 }] });
   |                                          ^^^ missing `b`
